### PR TITLE
add provide generated ecu.test reports step (#159)

### LINKED ISSUE #159: Add step to publish generated ecu.test reports
### Describe the feature request

As a plugin user I want to publish my generated reports to jenkins.

### Acceptance criteria

- [x] new step exists
- [ ]  step uploads all found generated reports as zip archives (one zip per generated report)
- [ ]  uploaded reports are available in an own sidebar entry in the pipeline view (like on the ecutest plugin)
- [x]  new step is documented
- [ ]  tests got extended on all test levels

### Additional Information
- this issue is just about the generic upload without any additional processing inside Jenkins (like process junit results)

### PR DESCRIPTION
**Pull request by MxEh-TT**
Closes #159

### CODE REVIEW

**Review by MartinGroscheTT — CHANGES_REQUESTED** [review]

File: src/main/groovy/de/tracetronic/jenkins/plugins/ecutestexecution/steps/ProvideGeneratedReportsStep.groovy [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    @DataBoundSetter
+    void setSelectedReportTypes(String selectedReportTypes) {
+        this.selectedReportTypes = (selectedReportTypes != null) ? selectedReportTypes : GenerateReportsStep.DescriptorImpl.REPORT_GENERATORS.join(", ")
+    }
+
+
+    protected ArrayList<String> processReport(File reportZip, String reportDirName, String outDirPath, TaskListener listener) {
+        ArrayList<String> reportPaths = []
+        selectedReportTypes.split(",\\s*").each { reportType ->
+            def folderEntryPaths = ZipUtil.getAllMatchingPaths(reportZip, "${reportType}*/**")
```
Comment (ErikRehmTT) [inline src/main/groovy/de/tracetronic/jenkins/plugins/ecutestexecution/steps/ProvideGeneratedReportsStep.groovy:47]: `${reportType}*`: I would give the power of matching with wildcard to the user

A check that the reportType is not `*` or `**` should be added and when found skipped

File: src/main/groovy/de/tracetronic/jenkins/plugins/ecutestexecution/steps/ProvideGeneratedReportsStep.groovy [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+            if (folderEntryPaths) {
+                def folderName = folderEntryPaths[0].substring(0, folderEntryPaths[0].indexOf('/'))
+                def outputFile = new File("${outDirPath}/${reportDirName}/${folderName}.zip")
+                outputFile.parentFile.mkdirs()
+                reportPaths.add(ZipUtil.recreateWithFilesOfType(reportZip, folderEntryPaths, outputFile))
+            }
```
Comment (ErikRehmTT) [inline src/main/groovy/de/tracetronic/jenkins/plugins/ecutestexecution/steps/ProvideGeneratedReportsStep.groovy:53]: This creates a zip with everything that matches. So if two reports match, both are in the same zip. But there should be s separate zip file for each report. Keep in mind, there can be reports in nested projects/subfolders

I would expect the the zip then contains the content of the report folder but NOT the report folder it self on toplevel

File: src/main/resources/de/tracetronic/jenkins/plugins/ecutestexecution/steps/ProvideGeneratedReportsStep/config.jelly [hunk 3]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<?jelly escape-by-default='true'?>
+<j:jelly xmlns:j="jelly:core" xmlns:f="/lib/form">
+    <f:description>${%step.description}
+        <b>${%step.compatNote}</b>
+    </f:description>
+    <f:advanced>
+        <f:entry title="${%selectedReportTypes.title}" description="${%selectedReportTypes.description}"
```
Comment (ErikRehmTT) [inline src/main/resources/de/tracetronic/jenkins/plugins/ecutestexecution/steps/ProvideGeneratedReportsStep/config.jelly:8]: this shouldn't be an advanced config option

File: src/main/resources/de/tracetronic/jenkins/plugins/ecutestexecution/steps/ProvideGeneratedReportsStep/config.properties [hunk 4]
```diff
@@ -0,0 +1,4 @@
+step.description=This step provides all custom generated ecu.test reports as job artifacts.
+step.compatNote=This feature is only available for ecu.test 2024.3 or higher!
+selectedReportTypes.title=Select Generated Reports
+selectedReportTypes.description=Generated report name that should be provided.
```
Comment (ErikRehmTT) [inline src/main/resources/de/tracetronic/jenkins/plugins/ecutestexecution/steps/ProvideGeneratedReportsStep/config.properties:4]: Add that it needs to be a comma separated list with the possibility to use wildcards

**Review by ErikRehmTT — CHANGES_REQUESTED** [review]
I don't like the sidebar namings: It should be `ecu.test Logs`, `ecu.test Reports` and `Generated ecu.test Reports`

**Review by ErikRehmTT — CHANGES_REQUESTED** [review]

**Review by ErikRehmTT — APPROVED** [review]

**Pull request merged**